src/util/safelywritablefile.cpp

```diff
@@ -263,6 +241,51 @@ bool SafelyWritableFile::commit() {
     return true;
 }
 
+void SafelyWritableFile::useTempFileWithPrefix() {
+    QString origFilePath = QFileInfo(m_origFileName).canonicalPath();
+    QString origFileCompleteBasename = QFileInfo(m_origFileName).completeBaseName();
+    QString origFileSuffix = QFileInfo(m_origFileName).suffix();
+    QString tempFileCompleteBasename = kSafelyWritableTempFilePrefix +
+            origFileCompleteBasename + '.' + origFileSuffix;
+    QString tempFileName = origFilePath + '/' + tempFileCompleteBasename;
+    m_tempFileName = std::move(tempFileName);
+}
+
+void SafelyWritableFile::useTempFileWithSuffix() {
+    QString tempFileName = m_origFileName + kSafelyWritableTempFileSuffix;
+    QFile origFile(m_origFileName);
+    if (!origFile.copy(tempFileName)) {
```

Comment:
This does not belong here, because the function name doe snot mention the copy

Reply:
So I guess we should rename the function name in this case. Because we need to copy the file for the `MetadataSourceTaglib` right?

Reply:
As I've mentioned this is moved back to the constructor now